docs/howto/use-additional-execute-contextmanager.rst

```diff
@@ -0,0 +1,47 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+
+Defining Additional Execute Context Manager
+===========================================
+
+Creating new context manager
+----------------------------
+
+Users can create their own execution context manager to allow context management on a higher level.
+To do so, one must define a new context manager in one of their files:
+
+    .. code-block:: python
+
+      import contextlib
+
+
+      @contextlib.contextmanager
+      def example_user_context_manager(task_instance, execution_context):
+          """
+          An example of a context manager that a user may provide.
+          """
+          in_context = True
+          try:
+              yield in_context
+          finally:
+              in_context = False
```

Comment:
Can we make this example more detailed? For example with authenticating or checking something before/after execution?

Reply:
We honestly should incorporate Evgeny's example into this somehow, open for ideas